tests/e2e/fine_grained_access/path_filtering_tests.py

```diff
@@ -0,0 +1,432 @@
+import common
+import sys
+import pytest
+
+
+def test_weighted_shortest_path_all_edge_types_all_labels_granted():
+    admin_connection = common.connect(username="admin", password="test")
+    user_connnection = common.connect(username="user", password="test")
+    common.execute_and_fetch_all(admin_connection.cursor(), "REVOKE LABELS * FROM user;")
+    common.execute_and_fetch_all(admin_connection.cursor(), "REVOKE EDGE_TYPES * FROM user;")
+    common.execute_and_fetch_all(admin_connection.cursor(), "GRANT READ ON LABELS * TO user;")
+    common.execute_and_fetch_all(admin_connection.cursor(), "GRANT READ ON EDGE_TYPES * TO user;")
+
+    total_paths_results = common.execute_and_fetch_all(
+        user_connnection.cursor(), "MATCH p=(n)-[r *wShortest (r, n | r.weight)]->(m) RETURN p;"
+    )
+    path_length_result = common.execute_and_fetch_all(
+        user_connnection.cursor(),
+        "MATCH p=(n:label0)-[r *wShortest (r, n | r.weight) path_length]->(m:label4) RETURN path_length;",
+    )
+
+    assert len(total_paths_results) == 16
+    assert path_length_result[0][0] == 20
```

Comment:
it would be good if you had the list of expected results in an array and then do something like this:
```
expected results = [...]
assert len(total_paths_results) = len(expected_results)
for res in total_paths_results:
  assert res in expected_results
```